xml/iso_template.py

```diff
@@ -1,21 +1,94 @@
 #!/usr/bin/env python3
 
-# This is the most basic code needed to load the template and write XML
-
 # From this directory, run: python iso_template.py
 
 
 from jinja2 import Environment, FileSystemLoader
 import yaml
 import re
 from lxml import etree
+import os
+
+TEMPLATE_FILE = './cioos_template.xml'
+
+
+def get_instruments_from_record(record):
+    ''' converts flat instrument variables to a nested dict
+        See test for example
+    '''
+    instruments = {}
+    for key, val in record.items():
+        matches_instrument = re.search('^instrument_(\d)+_?(\w+)?', key)
```

Comment:
In PyCharm I'm getting a  weak warning "PEP8 invalid escape sequence \d and \w"
Reading up on this briefly, it seems the recommendation is to either use a prefix for the raw string with r, or adding a second \ before each entry. Either will work from a PEP8 perspective. With those couple updates the tests still pass.
r'^instrument_(\d)+_?(\w+)?'
OR
'^instrument_(\\d)+_?(\\w+)?'
https://github.com/PyCQA/pycodestyle/issues/766
In theory it may eventually cause an issue even though it isn't presently.

Reply:
fixed, good to know!